Curs/src/Modules/new-evaluationModule.js

```diff
@@ -1,7 +1,38 @@
 interviewApp.newEvaluationModule = {
+
     init() {
-        app.innerHTML = interviewApp.NewEvaluationPage();
-        interviewApp.newEvaluationEvents(); // call newEvaluation events function
-        interviewApp.commonEvents(); // call commonEvents function which will load the login page
+        const newPromise = new Promise(function (resolve, reject) {
+            const xhttp = new XMLHttpRequest();
+
+            xhttp.onreadystatechange = function () {
+                if (this.readyState === 4) {
+                    if (this.status === 200) {
+                        var newEvalObject;
+                        try {
+                            newEvalObject = JSON.parse(xhttp.responseText);
+                        } catch (e) {
+                            reject('Something went wrong')
+                        }
+                        resolve(newEvalObject);
+                    } else {
+                        reject('Status wrong')
+                    }
+                }
+            }
+
+            xhttp.open('GET', 'src/Data/xhrNew.json', true)
+            xhttp.send()
+        });
+        newPromise.then((newObject) => {
+            app.innerHTML = interviewApp.NewEvaluationPage(newObject);
+            interviewApp.newEvaluationEvents(); // call newEvaluation events function
+            interviewApp.commonEvents(); // call commonEvents function which will load the login page
```

Comment:
Hmmm... I find it interesting that `interviewApp.commonEvents` loads the login page. Can you explain to me how that works and why it's needed?

Reply:
I feel a tricky question here but don't worry. I'm prepared. :D

I will explain you the things and how they work .

I created a commonEvents.js file where I added an eventListener on the logout button. 
When the user clicks it exists a function which will removeItem from sessionStorage and will take the user to the login page. Also, inside that function exists interviewApp.loginModule.init() which will initialize the login page.

This is how it works in big lines. :)

Reply:
Mkay, sounds reasonable.